inspirehep/records/indexer/tasks.py

```diff
@@ -0,0 +1,122 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2019 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+import logging
+from celery import shared_task
+from elasticsearch import NotFoundError
+from invenio_db import db
+from invenio_pidstore.models import PersistentIdentifier
+from sqlalchemy import tuple_
+from sqlalchemy.orm.exc import StaleDataError, NoResultFound
+
+from inspirehep.records.errors import MissingCitedRecordError
+from inspirehep.records.indexer.base import InspireRecordIndexer
+
+logger = logging.getLogger(__name__)
+
+
+def index_record_task(record):
+    pid_type = record.pid_type
+    pid_value = record["control_number"]
+
+    index_record.delay(
+        pid_value=pid_value,
+        pid_type=pid_type,
+        record_version=record.model.version_id,
+        deleted=record.get("deleted", False),
+    )
+    logger.info(f"Index record task sent for record {record.id}")
+
+
+def get_record(pid_type, pid_value, record_version=None):
+    logger.debug(f"Pulling record {pid_type}:{pid_value} on version {record_version}")
+    from inspirehep.records.api import InspireRecord
+
+    record = InspireRecord.get_record_by_pid_value(
+        pid_value=pid_value, pid_type=pid_type
+    )
+    if record_version and record.model.version_id < record_version:
+        logger.info(
+            f"Cannot pull record {pid_type}:{pid_value}in version {record_version}."
+            f"Current version: {record.model.version_id}."
+        )
+        raise StaleDataError()
+    return record
+
+
+@shared_task(ignore_result=False, bind=True)
+def process_references_for_record(
+    pid_type=None, pid_value=None, record_version=None, record=None
```

Comment:
This signature needs more checking if you want to keep everything nullable. Otherwise You should change it :boom: